app-backend/db/src/main/scala/com/azavea/rf/database/UserGroupRoleDao.scala

```diff
@@ -0,0 +1,94 @@
+package com.azavea.rf.database
+
+import com.azavea.rf.database.Implicits._
+import com.azavea.rf.datamodel.{UserGroupRole, GroupType, User}
+
+import doobie._, doobie.implicits._
+import doobie.postgres._, doobie.postgres.implicits._
+import doobie.util.transactor.Transactor
+import cats._, cats.data._, cats.effect.IO, cats.implicits._
+import io.circe._
+
+import java.util.UUID
+
+object UserGroupRoleDao extends Dao[UserGroupRole] {
+    val tableName = "user_group_roles"
+
+    val selectF = fr"""
+        SELECT
+            id, created_at, created_by,
+            modified_at, modified_by, is_active,
+            user_id, group_type, group_id, group_role
+        FROM
+    """ ++ tableF
+
+    def createF(ugr: UserGroupRole) =
+        fr"INSERT INTO" ++ tableF ++ fr"""(
+            id, created_at, created_by,
+            modified_at, modified_by, is_active,
+            user_id, group_type, group_id, group_role
+        ) VALUES (
+            ${ugr.id}, ${ugr.createdAt}, ${ugr.createdBy},
+            ${ugr.modifiedAt}, ${ugr.modifiedBy}, ${ugr.isActive},
+            ${ugr.userId}, ${ugr.groupType}, ${ugr.groupId}, ${ugr.groupRole}
+        )
+        """
+
+    // We don't want to support changes to roles beyond deactivation and promotion
+    def updateF(ugr: UserGroupRole, id: UUID, user: User) =
+        fr"UPDATE" ++ tableF ++ fr"""SET
+            modified_at = NOW(),
+            modified_by = ${user.id},
+            is_active = ${ugr.isActive},
+            group_role = ${ugr.groupRole}
```

Comment:
I can demote myself here -- should I be able to do that?

Reply:
Context: in another project users had this power and one client's only admin demoted himself and they wound up without any.

Reply:
Yeah, you shouldn't be able to do that. I'm unsure how much to put into access-control at this stage of the feature and was sort of kicking the can down the road on this until we had things more fleshed out. At the very least I think there should probably be comments about policies that will need to be enforced.

Reply:
Added a comment about it